src/AuthProvider.tsx

```diff
@@ -178,16 +197,16 @@ export const AuthProvider = (props: AuthProviderProps): JSX.Element => {
 
     const signoutRedirect = useCallback(
         userManager
-            ? (args?: SignoutRedirectArgs) => userManager.signoutRedirect(args).then(onSignoutRedirect)
+            ? (args?: SignoutRedirectArgs) => userManagerContext.signoutRedirect(args).then(onSignoutRedirect)
             : unsupportedEnvironment("signoutRedirect"),
-        [userManager, onSignoutRedirect],
+        [userManagerContext.signoutRedirect, onSignoutRedirect],
```

Comment:
it still depends on on `userManager`, see Line 180/199 (`userManager ? ...`)

Reply:
Same kind of problem here, though it's harmless to add it in this case. I can actually update these ternaries to be `userManagerContext?.signoutRedirect ? ...` instead.

Reply:
i am good with both